sources/System/Renderer/Renderer3DSystem.hpp

```diff
@@ -0,0 +1,49 @@
+//
+// Created by cbihan on 24/05/2021.
+//
+
+#pragma once
+
+#include <type_traits>
+#include "System/System.hpp"
+#include "Entity/Entity.hpp"
+#include "Component/Position/PositionComponent.hpp"
+#include "Component/Drawable/Drawable3DComponent.hpp"
+#include "Window.hpp"
+
+namespace BBM
+{
+	template <class T>
+	class Renderer3DSystem : public WAL::System
+	{
+	private:
+		//! @brief The class to render
+		RAY::Window &_window;
+	public:
+		//! @brief ctor
+		explicit Renderer3DSystem(RAY::Window &window)
+			: WAL::System({typeid(PositionComponent), typeid(Drawable3DComponent<T>)}),
+			_window(window)
+		{
+		}
+
+		//! @brief Update the corresponding component of the given entity
+		//! @param entity The entity to update.
+		//! @param dtime The delta time.
+		void onUpdate(WAL::Entity &entity, std::chrono::nanoseconds dtime) override
+		{
+			auto &comp = entity.getComponent<Drawable3DComponent<T>>();
+			auto &pos = entity.getComponent<PositionComponent>();
+
+			comp.member.setPosition({pos.getX(), pos.getY(), pos.getZ()});
```

Comment:
You should implement a conversion operator in the Vector3 class to simply pass the vector to this method.